core/src/OpenMPTarget/Kokkos_OpenMPTarget_Instance.hpp

```diff
@@ -51,14 +51,18 @@ namespace Kokkos {
 namespace Experimental {
 namespace Impl {
 
+enum class openmp_fence_is_static { yes, no };
+
 class OpenMPTargetInternal {
  private:
   OpenMPTargetInternal()                            = default;
   OpenMPTargetInternal(const OpenMPTargetInternal&) = default;
   OpenMPTargetInternal& operator=(const OpenMPTargetInternal&) = default;
 
  public:
-  void fence();
+  void fence(openmp_fence_is_static is_static = openmp_fence_is_static::no);
+  void fence(const std::string& name,
+             openmp_fence_is_static is_static = openmp_fence_is_static::no);
```

Comment:
same here as for threads, this is a new public overload. Might be better to delegate to an impl_funciton

Reply:
See: https://github.com/kokkos/kokkos/pull/3966#discussion_r666305635